frontend/src/router/index.ts

```diff
@@ -1,30 +1,72 @@
-import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router'
-import HomeView from '../views/HomeView.vue'
+import { MutationTypes } from '@/store/mutation-types';
+import { store } from '@/store/store';
+import { createRouter, createWebHistory, RouteRecordRaw } from 'vue-router';
+import HomeView from '../views/HomeView.vue';
 
 const routes: Array<RouteRecordRaw> = [
-  {
-    path: '/',
-    name: 'home',
-    component: HomeView
-  },
-  {
-    path: '/about',
-    name: 'about',
-    // route level code-splitting
-    // this generates a separate chunk (about.[hash].js) for this route
-    // which is lazy-loaded when the route is visited.
-    component: () => import(/* webpackChunkName: "about" */ '../views/AboutView.vue')
-  },
-  {
-    path: '/signup',
-    name: 'signup',
-    component: () => import('../views/SignUpView.vue')
-  }
-]
+    {
+        path: '/',
+        name: 'home',
+        component: HomeView,
+    },
+    {
+        path: '/about',
+        name: 'about',
+        // route level code-splitting
+        // this generates a separate chunk (about.[hash].js) for this route
+        // which is lazy-loaded when the route is visited.
+        component: () =>
+            import(/* webpackChunkName: 'about' */ '../views/AboutView.vue'),
+    },
+    {
+        path: '/signup',
+        name: 'signup',
+        component: () => import('../views/SignUpView.vue'),
+    },
+    {
+        path: '/wallet',
+        name: 'wallet',
+        component: () => import('../views/WalletView.vue'),
+    },
+    {
+        path: '/assets',
+        name: 'assets',
+        component: () => import('../views/AssetsView.vue'),
+    },
+    {
+        path: '/logout',
+        name: 'logout',
+        component: () => import('../views/LogoutView.vue'),
+        meta: {
+            hideNavBar: true,
+        },
+    },
+];
 
 const router = createRouter({
-  history: createWebHistory(process.env.BASE_URL),
-  routes
-})
+    history: createWebHistory(process.env.BASE_URL),
+    routes,
+});
 
-export default router
+router.beforeEach((to, from, next) => {
+    const token = sessionStorage.getItem('access_token'); 
+    if (token) {
+        const endpoint = '/auth/validate';
+        fetch(process.env.VUE_APP_API_URL + endpoint, {
+            method: 'POST',
+            headers: {
+                'Authorization': `Bearer ${token}`
+            }
```

Comment:
We can create a file (or more) to centralize the authenticated API calls. We can define the base API URL and the headers in just one place. A Library like [axios](https://axios-http.com/) could help.

Later we can also use the store `actions` to handle with API calls

Reply:
Let's talk about that. It seems interesting, in fact. About axios, I was trying not to include too many libraries that I found "unnecessary". 
As you are the js expert, feel free to give us hints

Reply:
I will create a task to do this while this PR is merged. Then we can address this feature easily. Thanks for the input.